### PR TITLE
fix: ValueError target column of train dataframe has NaN or Inf.

### PR DESCRIPTION
**Pull request by arima-tsukasa**
@AkiraUra 

Raises an error if there are more than a certain number of nan or inf value rows in the target column of the training data.

This PR is a part of the following PRs.

* [core: fix/target_column_has_nan_or_inf](https://github.com/sapientml/core/tree/fix/target_column_has_nan_or_inf)
* [loaddata: fix/target_column_has_nan_or_inf](https://github.com/sapientml/loaddata/tree/fix/target_column_has_nan_or_inf)

Please take a review on it.

### CODE REVIEW

**Comment by arima-tsukasa** [comment]
@AkiraUra 

We have taken additional measures based on the following PR.

* https://github.com/sapientml/core/pull/55

This fix assumes that you have applied the following changes:

* https://github.com/sapientml/core/pull/67

**Review by AkiraUra — COMMENTED** [review]
Thank you for your PR. I have some comments. Could you consider them?

File: sapientml/params.py [hunk 1]
```diff
@@ -514,12 +516,8 @@ def _check_single_dataframe(self, df: pd.DataFrame, target_columns: List[str], t
...
         return True
 
-    def _isnot_nan(self, df: pd.DataFrame, target_columns: list[str]) -> bool:
-        for target_column in target_columns:
-            target_data = df[target_column].replace([-np.inf, np.inf], np.nan)
-            if target_data.isna().any():
-                return False
-        return True
+    def _isnot_nan(self, df: pd.DataFrame, target_columns: list[str], limit: int) -> bool:
+        return not (len(df[df[target_columns].isin([np.inf, -np.inf, np.nan]).any(axis=1)]) > limit)
```
[the remark this answers is not in the document]
Comment (arima-tsukasa) [reply sapientml/params.py]: Thank you for pointing it out.
That code is more beautiful, I fixed it as you pointed out.

File: sapientml/params.py [hunk 2]
```diff
@@ -482,7 +482,9 @@ def _check_single_dataframe(self, df: pd.DataFrame, target_columns: List[str], t
             raise ValueError(f"{target_data_name} doesn't have target columns")
 
         # 1. Check NaN in target column
-        if set(target_columns).issubset(set(df.columns)) and (not self._isnot_nan(df, target_columns)):
+        if (target_data_name == "train") and not self._isnot_nan(df, target_columns, 10):
+            raise ValueError(f"target column of {target_data_name} dataframe has NaN or Inf.")
+        elif (target_data_name != "train") and not self._isnot_nan(df, target_columns, 0):
```
[the remark this answers is not in the document]
Comment (arima-tsukasa) [reply sapientml/params.py]: Fix to check test data only if target column exists.

File: sapientml/params.py [hunk 3]
```diff
@@ -482,7 +482,9 @@ def _check_single_dataframe(self, df: pd.DataFrame, target_columns: List[str], t
             raise ValueError(f"{target_data_name} doesn't have target columns")
 
         # 1. Check NaN in target column
-        if set(target_columns).issubset(set(df.columns)) and (not self._isnot_nan(df, target_columns)):
+        if (target_data_name == "train") and not self._isnot_nan(df, target_columns, 10):
+            raise ValueError(f"target column of {target_data_name} dataframe has NaN or Inf.")
```
[the remark this answers is not in the document]
Comment (arima-tsukasa) [reply sapientml/params.py]: Thank you for pointing it out.
I revised it as you indicated.

**Pull request merged**